cranelift/codegen/meta/src/shared/settings.rs

```diff
@@ -3,6 +3,35 @@ use crate::cdsl::settings::{SettingGroup, SettingGroupBuilder};
 pub(crate) fn define() -> SettingGroup {
     let mut settings = SettingGroupBuilder::new("shared");
 
+    settings.add_enum(
+        "regalloc",
+        r#"Register allocator to use with the MachInst backend.
+
+        This selects the register allocator as an option among those offered by the `regalloc.rs`
+        crate. Please report register allocation bugs to the maintainers of this crate whenever
+        possible.
+
+        Note: this only applies to target that use the MachInst backend. As of 2020-04-17, this
+        means the x86_64 backend doesn't use this yet.
+
+        Possible values:
+
+        - `backtracking` is a greedy, backtracking register allocator as implemented in
+        Spidermonkey's optimizing tier Ionmonkey. It may take more time to allocate registers, but
+        it should generate better code in general, resulting in better throughput of generate code.
+        - `backtracking_checked` is the backtracking allocator with additional self checks that may
+        take some time to run, and thus are disabled by default.
```

Comment:
s/are/is/ (sorry, nitpicky)

Reply:
"self check**s**" is multiple.

Reply:
reformulated to "thus these checks are", to remove all possible doubt